commit 1d2b8792acc9adab82f83d1d503ea535eca526b1
Author: Christoph Moench-Tegeder <cmt@FreeBSD.org>

    powerpc64le support for FreeBSD
    
    Adapted from:
      commit 192a67a0d79f89529eaac61ea9e865b7cb32365e
      Author: Piotr Kubaj <pkubaj@FreeBSD.org>
      Date:   Sat May 15 02:45:46 2021 +0000
    
         security/nss: fix build on powerpc64le
    
    upstream even got half way to what we need

diff --git lib/freebl/Makefile lib/freebl/Makefile
index 2b23fc438..9e53e607e 100644
--- lib/freebl/Makefile
+++ lib/freebl/Makefile
@@ -300,7 +300,7 @@ else ifeq ($(CPU_ARCH),x86)
 endif
 endif # Darwin
 
-ifeq ($(OS_TARGET),Linux)
+ifeq (,$(filter-out Linux FreeBSD, $(OS_TARGET)))
 ifeq ($(CPU_ARCH),x86_64)
     # Lower case s on mpi_amd64_common due to make implicit rules.
     ASFILES  = arcfour-amd64-gas.s mpi_amd64_common.s
@@ -336,7 +336,7 @@ ifdef USE_64
     PPC_ABI := $(shell $(CC) -dM -E - < /dev/null | awk '$$2 == "_CALL_ELF" {print $$3}')
     ifeq ($(PPC_ABI),2)
         ASFILES += sha512-p8.s
-    ifeq ($(OS_TEST),ppc64le)
+    ifeq (,$(filter-out powerpc64le ppc64le, $(OS_TEST)))
         EXTRA_SRCS += chacha20poly1305-ppc.c ppc-gcm-wrap.c
         ASFILES += chacha20-ppc64le.s ppc-gcm.s
         DEFINES += -DHAVE_PLATFORM_GCM
